lib/CRMS/RightsPredictor.pm

```diff
@@ -0,0 +1,140 @@
+package CRMS::RightsPredictor;
+
+# Predict copyright term for Commonwealth countries based on
+# author death date or publication date.
+# Used in Commonwealth project UI for populating and updating UI with
+# rights based on metadata and researcher-provided data.
+# Also used for New Year rights rollover.
+
+# This could be extended to non-Commonwealth countries.
+
+# TODO: this code relies heavily on publication/copyright date,
+# and as a result makes Bib API calls multiple times in the course
+# of a single review. Extending the Metadata module to use a local
+# database cache for volumes in queue would enhance responsiveness
+# an eliminate some unnecessary "paranoia" code.
+
+use strict;
+use warnings;
+use utf8;
+
+use Carp;
+use POSIX;
+
+sub new {
+  my ($class, %args) = @_;
+  my $self = bless {}, $class;
+  $self->{$_} = $args{$_} for keys %args;
+  Carp::confess "No Metadata object passed to RightsPredictor" unless $args{'record'};
+  return $self;
+}
+
+# Last year the work was/will be in copyright in country of publication.
+# Returns the following structure:
+#   my $ldc = {
+#     'year' => 1933, # 4-digit int or undef on error
+#     'desc' => 'how derived'
+#   };
+# or undef on error or not enough info.
+sub last_domestic_copyright
+{
+  my $self     = shift;
+  my $year     = shift || ''; # ADD or Pub entered by user
+  my $is_crown = shift; # Crown copyright founds
+
+  if ($year !~ m/^-?\d{1,4}$/) {
+    return { year => undef, desc => "unsupported date format '$year'" };
+  }
+  my $country = $self->{record}->country;
+  if ($country eq 'United Kingdom' || $country eq 'Canada' || $country eq 'Australia') {
```

Comment:
Consider separately extracting behavior for each country as a separate subroutine -- we can store references to those subroutines in a hash and look up the correct sub for each country, or return the default error from line 67 if we can't find that.